src/snaphu/_unwrap.py

```diff
@@ -332,6 +333,24 @@ def unwrap(  # type: ignore[no-untyped-def]
         ntiles=ntiles, tile_overlap=tile_overlap, nproc=nproc
     )
 
+    # Ensure that input & output datsets are not too large
+    check_dataset_sizes(
+        ntiles,
+        tile_overlap,
+        regrow_conncomps=regrow_conncomps,
+        single_tile_reoptimize=single_tile_reoptimize,
+        unw=unw,
+        conncomp=conncomp,
+    )
+    if mask is not None:
+        check_dataset_sizes(
+            ntiles,
+            tile_overlap,
+            regrow_conncomps=regrow_conncomps,
+            single_tile_reoptimize=single_tile_reoptimize,
+            mask=mask,
+        )
```

Comment:
What about the `igram` and `corr` input datasets? Should we check their sizes as well?

There's an earlier [check](https://github.com/isce-framework/snaphu-py/blob/afe62434dfeddf9875dadf09784c412228f38d5c/src/snaphu/_unwrap.py#L304-L307) that ensures that all of the input & output arrays have the same shape, so actually I think it's sufficient to just check `igram`'s shape.

```suggestion
    # Ensure that the dataset (and tile) dimensions are not too large.
    check_dataset_sizes(
        ntiles,
        tile_overlap,
        regrow_conncomps=regrow_conncomps,
        single_tile_reoptimize=single_tile_reoptimize,
        igram=igram,
    )
```